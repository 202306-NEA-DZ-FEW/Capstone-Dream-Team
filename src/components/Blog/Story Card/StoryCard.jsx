import React from "react";

export default function StoryCard() {
    return (
        <div className='container mx-auto p-4'>
            {/* Story Card */}
            <div className='relative rounded-lg flex flex-col md:flex-row items-center md:shadow-xl md:h-72 mx-2'>
                {/* Left Side (Image) */}
                <div className='z-0 order-1 md:order-2 relative w-full md:w-2/5 h-80 md:h-full overflow-hidden rounded-lg md:rounded-none md:rounded-r-lg'>
                    {/* Background Image */}
                    <div
                        className='absolute left-0 top-0 inset-0 w-full h-full object-center bg-opacity-30 bg-cover'
                        style={{
                            backgroundImage:
                                "url(./images/fatoom_in_yemen.png)",
                            backgroundBlendMode: "multiply",
                        }}
                    ></div>

                    {/* Hidden Content on Mobile */}
                    <div className='md:hidden absolute inset-0 h-full p-6 pb-6 flex flex-col-reverse justify-start items-start bg-gradient-to-b from-transparent via-transparent to-gray-900'>
                        <h3 className='w-full font-bold text-2xl text-teal-500 leading-tight mb-2'>
                            Fatoom in Yemen
                        </h3>
                    </div>

                    {/* Diagonal Polygon Shape (Visible on Desktop) */}
                    <svg
                        className='hidden md:block absolute inset-y-0 h-full w-24 fill-current text-white -ml-12'
                        viewBox='0 0 100 100'
                        preserveAspectRatio='none'
                    >
                        <polygon points='50,0 100,0 50,100 0,100' />
                    </svg>
                </div>

                {/* Right Side (Text Content) */}
                <div className='z-10 order-2 md:order-1 w-full h-full md:w-3/5 flex items-center -mt-6 md:mt-0'>
                    <div className='p-8 md:pr-18 md:pl-14 md:py-12 mx-2 md:mx-0 h-full bg-white rounded-lg md:rounded-none md:rounded-l-lg shadow-xl md:shadow-none'>
                        {/* Title (Visible on Desktop) */}
                        <h3 className='hidden md:block font-bold text-2xl text-teal-500 mb-2'>
                            Fatoom in Yemen
                        </h3>

                        {/* Main Text Content */}
                        <p className='text-gray-600 text-justify text-base line-clamp-4'>
                            Baby Fatoom and her family were driven from their
                            home in Yemen by conflict: the number one cause of
                            hunger. She was diagnosed with Severe Acute
                            Malnutrition (SAM) at just nine months old. As we do
                            with so many mothers across the world – we made sure
                            Fatoom’s mother Zainab got the nutrition she needed
                            to breastfeed. In places like Yemen and around the
                            world, nearly 50% of deaths among children under
                            five are caused by hunger.
                        </p>

                        {/* Read More Link */}
                        <div className='flex justify-between mt-3'>
                            <a
                                className='flex items-baseline  text-teal-500 hover:text-teal-900 focus:text-teal-900'
                                href=''
                            >
                                <span>Read More</span>
                                <span className='text-xs ml-1'>&#x279c;</span>
                            </a>
                            <div className='flex items-baseline text-sm ml-1 text-gray-500'>
                                <span>26/10/2023</span>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}
