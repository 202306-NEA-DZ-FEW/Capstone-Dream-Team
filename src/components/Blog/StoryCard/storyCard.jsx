import { useTranslation } from "next-i18next";
import React, { useEffect, useState } from "react";

export default function StoryCard({ article, index, language }) {
    const { t } = useTranslation("common");
    const [isPair, setIsPair] = useState(true);
    useEffect(() => {
        if (index % 2 == 0) setIsPair(true);
        else setIsPair(false);
    }, []);
    return (
        <div className='container mx-auto md:py-5 md:px-18 p-4 lg:px-20 lg:py-5'>
            {/* Story Card */}
            <div
                className={`relative rounded-lg flex flex-col md:flex-row items-center md:shadow-xl md:h-80 mx-2 ${
                    isPair ? "md:flex-row" : "md:flex-row-reverse"
                }`}
            >
                {/* Left Side (Image) */}
                <div
                    className={`z-0 relative w-full md:w-2/5 h-96 md:h-full overflow-hidden rounded-lg ${
                        !isPair
                            ? "md:rounded-none md:rounded-r-lg"
                            : "md:rounded-l-lg"
                    } md:rounded-none`}
                >
                    {/* Background Image */}
                    <div
                        className='absolute left-0 top-0 inset-0 w-full h-full object-center bg-opacity-30 bg-cover'
                        style={{
                            backgroundImage: "url(images/fatoom_in_yemen.png)",
                            backgroundBlendMode: "multiply",
                        }}
                    ></div>
                    {/* Hidden Content on Mobile */}
                    <div className='md:hidden absolute inset-0 h-full p-6 pb-6 flex flex-col-reverse justify-start items-start bg-gradient-to-b from-transparent to-gray-900'>
                        <h3 className='w-full font-bold text-2xl text-teal-500 leading-tight mb-2'>
                            {article.title}
                        </h3>
                    </div>
                    {/* Diagonal Polygon Shape (Visible on Desktop) */}
                    <svg
                        className={`hidden md:block absolute inset-y-0 h-full w-24 fill-current text-white ${
                            !isPair ? "-ml-12 " : "-mr-12 right-0"
                        }`}
                        viewBox='0 0 100 100'
                        preserveAspectRatio='none'
                    >
                        <polygon points='50,0 100,0 50,100 0,100' />
                    </svg>
                </div>

                {/* Right Side (Text Content) */}
                <div className='z-0 w-full h-full md:w-3/5 flex items-center -mt-6 md:mt-0'>
                    <div className='p-8 md:pr-18 md:pl-14 md:py-12 mx-2 md:mx-0 h-full bg-white rounded-lg md:rounded-none shadow-xl md:shadow-none'>
                        {/* Title (Visible on Mobile) */}
                        <h3 className='hidden md:block font-bold text-2xl text-teal-500 mb-2 line-clamp-1 max-h-8'>
                            {article.title}
                        </h3>
                        {/* Main Text Content */}
                        <p className='text-gray-600 text-justify text-base md:line-clamp-6 line-clamp-3'>
                            {article.content}
                        </p>

                        {/* Read More Link */}
                        <div
                            className={`flex justify-between ${
                                language === "ar" ? "flex-row-reverse" : "flex"
                            } mt-3`}
                        >
                            <div className='flex items-baseline text-sm ml-1 text-gray-500'>
                                <span>{article.publish_date}</span>
                            </div>{" "}
                            <a
                                className='flex items-baseline text-teal-500 hover:text-teal-900 focus:text-teal-900'
                                href=''
                            >
                                <span>{t("readMore")}</span>
                                {language === "ar" ? (
                                    <span className='text-xs ml-1'>
                                        &#x2190;
                                    </span>
                                ) : (
                                    <span className='text-xs ml-1'>
                                        &#x279c;
                                    </span>
                                )}
                            </a>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}
