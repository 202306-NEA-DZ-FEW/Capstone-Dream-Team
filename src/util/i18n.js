import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
