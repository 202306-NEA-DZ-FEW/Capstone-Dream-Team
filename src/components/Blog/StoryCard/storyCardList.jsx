import React from "react";

import StoryCard from "./storyCard";

export default function StoryCardList({ stories, numToShow }) {
    const displayedStories = stories.slice(0, numToShow);

    return (
        <div>
            {displayedStories.map((story, index) => (
                <div key={index}>
                    <StoryCard story={story} index={index} />
                </div>
            ))}
        </div>
    );
}
