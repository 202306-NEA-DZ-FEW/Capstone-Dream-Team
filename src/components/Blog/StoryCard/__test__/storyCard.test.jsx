import { I18nextProvider } from "react-i18next";
import renderer from "react-test-renderer";

import StoryCard from "../storyCard";
import i18n from "../../../../util/i18n";
const article = {
    title: "Sample Title 1",
    content: "Sample Content 1",
    publish_date: "2023-10-31",
};

it("renders correctly", () => {
    const tree = renderer
        .create(
            <I18nextProvider i18n={i18n}>
                <StoryCard language='en' article={article} index={1} />
            </I18nextProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
