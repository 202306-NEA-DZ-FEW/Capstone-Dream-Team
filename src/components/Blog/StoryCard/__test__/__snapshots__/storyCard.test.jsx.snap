// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl md:px-24 lg:px-24 lg:py-12"
>
  <div
    className="flex flex-col items-center justify-between lg:flex-row"
  >
    <div
      className="mb-10 lg:max-w-lg lg:px-5 lg:mb-0"
    >
      <div
        className="max-w-xl mb-6"
      >
        <h2
          className="dark:text-white max-w-lg mb-6 font-sans text-3xl font-bold tracking-tight text-gray-900 sm:text-4xl sm:leading-none line-clamp-2"
        >
          Sample Title 1
        </h2>
        <p
          className="text-base dark:text-white sm:text-xl text-gray-700 md:text-lg line-clamp-4"
        >
          Sample Content 1
        </p>
        <div>
          <a
            className="flex "
            href="blogs/story/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <p
              className="text-base flex text-teal-500 font-bold no-underline hover:underline py-4"
            >
              <span />
              <svg
                className="w-4 h-4 ml-1 mt-1"
                fill="none"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 12h14"
                />
                <path
                  d="M12 5l7 7-7 7"
                />
              </svg>
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      className="relative lg:w-1/2"
    >
      <img
        alt=""
        className="object-cover w-full h-full rounded shadow-lg sm:h-96"
        src="/images/fatoom_in_yemen.png"
      />
    </div>
  </div>
</div>
`;
